### DISCUSSION TITLE
Can you parse variants from the result of a method_call?

### DISCUSSION THREAD

**Discussion by catpreneur** [diwic/dbus-rs#313] [General]
I'm aware of the code examples here https://github.com/diwic/dbus-rs/blob/master/dbus/examples/argument_guide.md#variants and here https://github.com/diwic/dbus-rs/issues/225 

However, I can't use proxy.get() e.g. `p.get("org.mpris.MediaPlayer2.Player", "Metadata").unwrap();` because the method I need to call needs input, and I don't think you can pass parameters using a `.get()` method.

Seems like I can't do 
```
    let infoinfo: HashMap<String, arg::Variant<Box<dyn RefArg>>> = pr.method_call(...).unwrap();
```

as i get

```
   |
72 | ...Map<String, arg::Variant<Box<dyn RefArg>>> = pr2.method_call(...
   |                                                     ^^^^^^^^^^^ the trait `ReadAll` is not implemented for `HashMap<std::string::String, dbus::arg::Variant<std::boxed::Box<dyn RefArg>>>`

```

**Comment by diwic** [comment]
The return value from `method_call` is always a tuple, so just untuple it like this:

    let (infoinfo,): (arg::PropMap,) = pr.method_call(...)

**Reply by catpreneur** [reply]
Thank you!
